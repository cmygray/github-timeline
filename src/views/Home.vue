<template>
  <v-container tag="section" class="home">
    <h1>GitHub timeline</h1>
    <form @submit.prevent="handleSubmit">
      <v-text-field
        v-model="userId"
        autofocus
        label="GitHub ID"
        placeholder="cmygray"
      />
      <v-btn type="submit" block color="primary">Generate</v-btn>
    </form>
  </v-container>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

@Component
export default class Home extends Vue {
  private userId: string = '';

  private handleSubmit() {
    this.$router.push({
      name: 'repos.index',
      query: {
        userId: this.userId,
      },
    });
  }
}
</script>
