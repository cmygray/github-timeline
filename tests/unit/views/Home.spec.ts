import { createLocalVue, mount } from '@vue/test-utils';
import Vue from 'vue';
import Vuetify from 'vuetify';

import Home from '@/views/Home.vue';

Vue.use(Vuetify);

const localVue = createLocalVue();
// @ts-ignore-next-line
const vuetify = new Vuetify();

describe('Home.vue', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = mount(Home, {
      localVue,
      vuetify
    });
  });

  it('renders as expected', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('input', () => {
    let input: any;

    beforeEach(function() {
     input = wrapper.find('input');
    });

    it('should have empty string as default value', function() {
      expect(input.element.value).toEqual('')
    });

    it('should show and change userId data', function() {
      wrapper.setData({ userId: 'foo' });

      expect(input.element.value).toEqual('foo');

      input.setValue('bar');

      expect(wrapper.vm.userId).toEqual('bar');
    });
  });

  describe('#handleSubmit', () => {
    const USER_ID = 'cmygray';
    const NEXT_ROUTE_NAME = 'repos.index';

    beforeEach(function() {
      wrapper = mount(Home, {
        localVue,
        vuetify,
        data: () => ({
          userId: USER_ID
        }),
        mocks: {
          $router: {
            push: jest.fn()
          }
        }
      })
    });

    it('should push new route and query string to router', function() {
      wrapper.vm.handleSubmit();

      expect(wrapper.vm.$router.push).toBeCalledWith({
        name: NEXT_ROUTE_NAME,
        query: { userId: USER_ID }
      })
    });
  });
});
