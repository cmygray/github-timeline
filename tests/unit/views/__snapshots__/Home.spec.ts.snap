// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue renders as expected 1`] = `
<section class="container home">
  <h1>GitHub timeline</h1>
  <form>
    <div class="v-input v-input--is-focused theme--light v-text-field v-text-field--placeholder primary--text">
      <div class="v-input__control">
        <div class="v-input__slot">
          <div class="v-text-field__slot"><label aria-hidden="true" class="v-label v-label--active theme--light primary--text" style="left: 0px; position: absolute;">GitHub ID</label><input aria-label="GitHub ID" autofocus="autofocus" placeholder="cmygray" type="text"></div>
        </div>
        <div class="v-text-field__details">
          <div class="v-messages theme--light primary--text"><span name="message-transition" tag="div" class="v-messages__wrapper"></span></div>
        </div>
      </div>
    </div> <button type="submit" class="v-btn v-btn--block v-btn--contained theme--light v-size--default primary"><span class="v-btn__content">Generate</span></button>
  </form>
</section>
`;
